src/main/java/org/folio/service/event/handler/processor/EventProcessor.java

```diff
@@ -11,74 +10,64 @@
 import org.apache.commons.collections4.map.CaseInsensitiveMap;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
-import org.folio.kafka.AsyncRecordHandler;
 import org.folio.persist.AbstractRepository;
 import org.folio.service.event.InventoryEventType;
 
 import io.vertx.core.Future;
 import io.vertx.core.json.JsonObject;
-import io.vertx.kafka.client.consumer.KafkaConsumerRecord;
 import lombok.RequiredArgsConstructor;
 
-public abstract class UpdateEventAbstractHandler<T> implements AsyncRecordHandler<String, String> {
+public abstract class EventProcessor<T> {
   private static final Logger log = LogManager.getLogger(MethodHandles.lookup().lookupClass());
 
-  private final InventoryEventType supportedEventType;
+  protected final InventoryEventType supportedEventType;
 
-  protected UpdateEventAbstractHandler(InventoryEventType supportedEventType) {
+  protected EventProcessor(InventoryEventType supportedEventType) {
     this.supportedEventType = supportedEventType;
   }
 
-  @Override
-  public Future<String> handle(KafkaConsumerRecord<String, String> event) {
-    final String eventKey = event.key();
-    log.info("handle:: received event {}", eventKey);
+  public Future<String> run(String eventKey, CaseInsensitiveMap<String, String> headers,
+    JsonObject payload) {
 
-    return processEvent(event)
+    log.info("run:: received event {}", eventKey);
+
+    return processEvent(headers, payload)
       .onSuccess(r -> log.info("handle:: event {} processed successfully", eventKey))
       .onFailure(t -> log.error("handle:: failed to process event", t))
       .map(eventKey);
   }
 
-  protected Future<List<T>> processEvent(KafkaConsumerRecord<String, String> event) {
-    JsonObject payload = new JsonObject(event.value());
+  private Future<List<T>> processEvent(CaseInsensitiveMap<String, String> headers,
```

Comment:
Since all of the parameters of this method come from an instance of KafkaConsumerRecord, can we replace them with one parameter of type KafkaConsumerRecord?